File: src/main/java/com/study/platform/global/kafka/KafkaMessagePublisher.java
package com.study.platform.global.kafka;

import java.util.concurrent.CompletableFuture;

public interface KafkaMessagePublisher {

    CompletableFuture<Void> publish(String topic, String key, String payload);

    default CompletableFuture<Void> publish(String topic, String payload) {
        return publish(topic, null, payload);
    }
}

File: domain/src/main/kotlin/com/study/platform/global/kafka/KafkaMessagePublisher.kt
package com.study.platform.global.kafka

import java.util.concurrent.CompletableFuture

interface KafkaMessagePublisher {
    fun publish(topic: String, key: String?, payload: String): CompletableFuture<Void>

    fun publish(topic: String, payload: String): CompletableFuture<Void> =
        publish(topic, null, payload)
}
